tests/contributing.py

```diff
@@ -304,6 +308,61 @@ def validate_transformer_code_style(transformer):
     return not bool(errors)
 
 
+def validate_transformer_unit_tests(transformer):
+    """Validate the unit tests of a transformer.
+
+    This function finds the module where the unit tests of the transformer
+    have been implemented and runs them using ``pytest``, capturing the code
+    coverage of the tests (how many lines of the source code are executed
+    during the tests).
+
+    Args:
+        transformer (string or rdt.transformers.BaseTransformer):
+            The transformer to validate.
+
+    Returns:
+        float:
+            A ``float`` value representing the test coverage where 1.0 is 100%.
+    """
+    if not inspect.isclass(transformer):
+        transformer = load_transformer(transformer)
+
+    source_location = inspect.getfile(transformer)
+    test_location = get_test_location(transformer)
+    module_name = getattr(transformer, '__module__', None)
+
+    print(f'Validating source file {source_location}\n')
+
+    pytest_run = f'-v --disable-warnings --no-header {test_location}'
+    pytest_run = pytest_run.split(' ')
+
+    cov = coverage.Coverage(source=[module_name])
+
+    cov.start()
+    pytest_output = pytest.main(pytest_run)
+    cov.stop()
+
+    if pytest_output is pytest.ExitCode.OK:
+        print('\nSUCCESS: The unit tests passed.')
+    else:
+        print('\nERROR: The unit tests failed.')
+
+    score = cov.report(show_missing=True)
+    rounded_score = round(score / 100, 3)
+    if rounded_score < 1.0:
+        print(f'\nERROR: The unit tests only cover {round(score, 3)}% of your code.')
+    else:
+        print(f'\nSUCCESS: The unit tests cover {round(score, 3)}% of your code.')
+
+    cov.html_report()
+    print('\nFull coverage report here:\n')
+    coverage_name = module_name.replace('.', '_')
+    export_dir = Path('htmlcov') / f'{coverage_name}_py.html'
+    print(export_dir.absolute().as_uri())
```

Comment:
Minor thing: This will print the full path as a URI, so it is clickable when the output is shown in a command line.
However, in a jupyter notebook this will be a plain string, which will not be clickable.

Could we move this into a function that detects whether we are in jupyter environment or not and, if we are, prints an HTML link that is clickable?

Reply:
Actually that can't be done because `jupyter-notebook` blocks the `localfile` links. I have attempted several approaches but those end up generating a `403` error code where you can't access the `css` files and others.

After a `1-1` discussion we decided to just print the path to where the file is expected and let the user copy and paste it in his browser.